### PR TITLE
[clkmgr] Minor documentation updates

### PR DESCRIPTION
**Pull request by tjaychen**
Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by zi-v** [comment]
Thanks Tim, LGTM

File: hw/ip/clkmgr/doc/_index.md [hunk 1]
```diff
@@ -231,12 +231,12 @@ There is currently no support in hardware to dynamically synchronize a threshold
...
-| -------------                | ---------------------  | ----------------- | ----------------------------| ----------------------- |
-| Life cycle transition        | `lc_ctrl_pkg::On`      | Don't care        | Don't care                  | Controlled by `lc_ctrl` |
-| Internal Clocks              | `lc_ctrl_pkg::Off`     | `kMultiBit4False` | Don't care                  | All                     |
-| Software external high speed | `lc_ctrl_pkg::Off`     | `kMultiBit4True`  | `kMultiBit4True`            | TEST_UNLOCKED, RMA      |
-| Software external low speed  | `lc_ctrl_pkg::Off`     | `kMultiBit4True`  | `kMultiBit4False`           | TEST_UNLOCKED, RMA      |
+| Mode                                            | `lc_clk_byp_req_i`     | `extclk_ctrl.sel` | `extclk_ctrl.hi_speed_sel`  | life cycle state        |
+| -------------                                   | ---------------------  | ----------------- | ----------------------------| ----------------------- |
+| Life cycle transition from RAW and TEST* states | `lc_ctrl_pkg::On`      | `kMultiBit4False` | Don't care                  | Controlled by `lc_ctrl` |
```
Comment (matutem) [inline hw/ip/clkmgr/doc/_index.md:236]: I see, you are probably right: if lc_clk_byp_req_i and sw requests external clock things can become confusing. Does this mean we cannot do lc_ctrl transitions once external clock is selected? Could that prevent TEST* transitions in some cases? For example, if the clocks are unstable and we use SW extclk to bypass the oscillators we would not be able to run LC transitions?

Comment (tjaychen) [reply hw/ip/clkmgr/doc/_index.md:236]: ah sorry @matutem, let me add you to the thread, it will be clearer.

Comment (matutem) [reply hw/ip/clkmgr/doc/_index.md:236]: We can leave aside the weird case I am concerned about.

**Review by matutem — APPROVED** [review]
LGTM

**Pull request merged**